### PR TITLE
✨ Add LIKE tx for arweave fee feature

### PR DESCRIPTION
**Pull request by williamchong**
Allow user to pay LIKE in a transaction, and use that transaction to tell our API to upload files to arweave
Requires 2 signing for now

### CODE REVIEW

File: components/IscnRegisterForm.vue [hunk 1]
```diff
@@ -343,17 +347,25 @@ export default class IscnRegisterForm extends Vue {
 
   @Watch('payload', { immediate: true, deep: true })
   change() {
-    this.debouncedCalculateTotalFee()
+    this.debouncedCalculateISCNFee()
+  }
+
+  @Watch('uploadArweaveId')
+  chang(newId: string) {
```
Comment (nwingt) [inline components/IscnRegisterForm.vue:354]:
```suggestion
  handleUploadArweaveIDChange(newId: string) {
```

File: components/IscnRegisterForm.vue [hunk 2]
```diff
@@ -343,17 +347,25 @@ export default class IscnRegisterForm extends Vue {
...
   @Watch('error')
   showWarning(errormsg: any) {
     if (errormsg) this.isOpenWarningSnackbar = true
   }
 
-  mounted() {
+  async mounted() {
+    this.uploadStatus = 'Loading'
+    await this.estimateArweaveFee();
+    // Total Fee needs Arweave fee to calculate
```
Comment (nwingt) [inline components/IscnRegisterForm.vue:366]:
```suggestion
    // ISCN Fee needs Arweave fee to calculate
```

File: components/IscnRegisterForm.vue [hunk 3]
```diff
@@ -459,40 +481,88 @@ export default class IscnRegisterForm extends Vue {
...
+          headers: {
+            'Content-Type': 'multipart/form-data',
+          },
+        },
+      );
+      this.uploadArweaveId = arweaveId;
+      if (LIKE) this.arweaveFee = new BigNumber(LIKE);
+      this.arweaveFeeTargetAddress = address;
+    } catch (err) {
+      console.error(err);
```
Comment (nwingt) [inline components/IscnRegisterForm.vue:526]:
```suggestion
      // TODO: Handle error
      console.error(err);
```

File: components/IscnRegisterForm.vue [hunk 4]
```diff
@@ -459,40 +481,88 @@ export default class IscnRegisterForm extends Vue {
...
+    }
+  }
+
+  async sendArweaveFeeTx(): Promise<string> {
+    if (!this.signer) throw new Error('SIGNER_NOT_INITED');
+    if (!this.arweaveFeeTargetAddress) throw new Error('TARGET_ADDRESS_NOT_SET');
+    this.uploadStatus = 'Waiting for signature'
```
Comment (nwingt) [inline components/IscnRegisterForm.vue:533]: Should localize

File: components/IscnRegisterForm.vue [hunk 5]
```diff
@@ -459,40 +481,88 @@ export default class IscnRegisterForm extends Vue {
...
+        {
+          headers: {
+            'Content-Type': 'multipart/form-data',
+          },
+        },
+      );
+      this.uploadArweaveId = arweaveId;
+      this.$emit('arweaveUploaded', { arweaveId })
+    } catch (err) {
+      console.error(err);
```
Comment (nwingt) [inline components/IscnRegisterForm.vue:558]:
```suggestion
      // TODO: Handle error
      console.error(err);
```

File: server/arweave/index.ts [hunk 6]
```diff
@@ -24,24 +25,35 @@ export async function getArIdFromHashes(ipfsHash: string) {
...
   }
   const transaction = await arweave.createTransaction({ data: buffer }, jwk)
   const { reward } = transaction;
   return {
     AR: arweave.ar.winstonToAr(reward),
-  }
+  };
+}
+
+export async function converARPriceToLIKE(ar: string, { margin = 0.05, decimal = 0 } = {}) {
```
Comment (nwingt) [inline server/arweave/index.ts:49]:
```suggestion
export async function convertARPriceToLIKE(ar: string, { margin = 0.05, decimal = 0 } = {}) {
```

File: server/arweave/router.ts [hunk 7]
```diff
@@ -10,20 +15,57 @@ router.post('/estimate', multer().single('file'), async (req, res, next) => {
       res.sendStatus(400);
       return
     }
-    const price = await estimatePrice(req.file);
-    res.json(price);
+    const { AR, arweaveId } = await estimateARPrice(req.file);
+    const LIKE = await converARPriceToLIKE(AR);
```
Comment (nwingt) [inline server/arweave/router.ts:19]:
```suggestion
    const LIKE = await convertARPriceToLIKE(AR);
```

**Review by nwingt — CHANGES_REQUESTED** [review]

**Comment by nwingt** [comment]
Please rebase likecoin:feature/arweave to likecoin:master

**Comment by williamchong** [comment]
Since the first branch out rebase would cause too many conflicts, so I had been using merge

**Pull request merged**